skfda/ml/classification/_logistic_regression.py

```diff
@@ -0,0 +1,193 @@
+from __future__ import annotations
+
+from typing import Callable, Tuple
+
+import numpy as np
+from sklearn.base import BaseEstimator, ClassifierMixin
+from sklearn.linear_model import LogisticRegression as mvLogisticRegression
+from sklearn.utils.validation import check_is_fitted
+
+from ..._utils import _classifier_get_classes
+from ...representation import FDataGrid
+from ...representation._typing import NDArrayAny, NDArrayInt
+
+
+class LogisticRegression(
+    BaseEstimator,  # type: ignore
+    ClassifierMixin,  # type: ignore
+):
+    r"""Logistic Regression classifier for functional data.
+
+    This class implements the sequential “greedy” algorithm
+    for functional logistic regression proposed in
+    :footcite:ts:`bueno++_2021_functional`.
+
+    .. warning::
+        For now, only binary classification for functional
+        data with one dimensional domains is supported.
+
+    Args:
+        p: number of points (and coefficients) to be selected by
+        the algorithm.
+
+    Attributes:
+        classes\_: A list containing the name of the classes
+        points\_: A list containing the selected points.
+        coef\_: A list containing the coefficient for each selected point.
+        intercept\_: Independent term.
+
+    Examples:
+        >>> from numpy import array
+        >>> from skfda.datasets import make_gaussian_process
+        >>> from skfda.ml.classification import LogisticRegression
+        >>> fd1 = make_gaussian_process(
+        ...         n_samples=50,
+        ...         n_features=100,
+        ...         noise=0.7,
+        ...         random_state=0,
+        ... )
+        >>> fd2 = make_gaussian_process(
+        ...         n_samples=50,
+        ...         n_features = 100,
+        ...         mean = array([1]*100),
+        ...         noise = 0.7,
+        ...         random_state=0
+        ... )
+        >>> fd = fd1.concatenate(fd2)
+        >>> y = 50*[0] + 50*[1]
+        >>> lr = LogisticRegression()
+        >>> _ = lr.fit(fd[::2], y[::2])
+        >>> lr.coef_.round(2)
+        array([[ 1.28,  1.17,  1.27,  1.27,  0.96]])
+        >>> lr.points_.round(2)
+        array([ 0.11,  0.06,  0.07,  0.03,  0.  ])
+        >>> lr.score(fd[1::2],y[1::2])
+        0.94
+
+        References:
+            .. footbibliography::
+
+    """
+
+    def __init__(
+        self,
+        p: int = 5,
+    ) -> None:
+
+        self.p = p
+
+    def fit(  # noqa: D102
+        self,
+        X: FDataGrid,
+        y: NDArrayAny,
+    ) -> LogisticRegression:
+
+        X, classes, y_ind = self._argcheck_X_y(X, y)
+
+        self.classes_ = classes
+
+        n_samples = len(y)
+        n_features = len(X.grid_points[0])
+
+        selected_indexes = np.zeros(self.p, dtype=np.intc)
+
+        mvlr = mvLogisticRegression()  # multivariate logistic regression
```

Comment:
Note that by default logistic regression in scikit-learn has a L2 regularization. If that is intended make it explicit, otherwise pass the regularization used by the original article.